### PR TITLE
Add debuginfo build target

### PR DESCRIPTION
**Pull request by nhorman**
In the webinar we are currently producing on debugging openssl applications, we talk about ways to allow debugable binaries without having to ship all the debug DWARF information to production systems.

Add an optional target to do that DWARF separation to aid users

##### Checklist
- [x] documentation is added or updated

### CODE REVIEW

**Comment by t8m** [comment]
Would you please consider extending some CI job with `make debuginfo` to check it?

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
@nhorman I think this is worth adding CHANGES.md entry.

File: CHANGES.md
Comment (t8m) [inline CHANGES.md:32]: Either `make` or `Makefile`, also s/platofms/platforms/

File: CHANGES.md
Comment (t8m) [inline CHANGES.md:33]: s/separated/separate/ sounds better to me, also please add a dot at the end of the sentence.

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
ping @openssl/committers for second review

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Merged to the master branch. Thank you.

**Pull request closed without merging**
